### DISCUSSION TITLE
Minio error 504 time out for large bucket.

### DISCUSSION THREAD

**Discussion by felipejareta** [minio/minio#15832] [Q&A]
Hi everybody.

I have a Minio hosted and running on openshift with no tuning or special configuration.
In this Minio, there is a bucket that have +570k small files, and everytime that I trie to access this folder I see "This operation is taking longer than expected..." and then 504 error in header page.

I'm not specialized in Minio, is there anybody that knows about some tuning configuration for this?

Regards.

**Comment by vadmeste** [comment]
> and everytime that I trie to access this folder

What do you mean exactly by accessing this folder ? Also what is the hierarchy of files stored in that bucket ?

**Reply by felipejareta** [reply]
I have a bucket and inside it there are another folders storing many files. 
The structure is:
Bucket name > Folder of Images > Many Folders with files inside (The slowness is here)

Depending of counting files, to open it can lasts 522 seconds.

**Reply by klauspost** [reply]
The solution is not to have 570k small files in the same prefix on a small setup.

https://min.io/docs/minio/linux/administration/object-management.html#object-organization-and-planning

**Comment by opslinux** [comment]
The same wrong happened to me, any news?

**Reply by marktheunissen** [reply]
Hi, did you read https://min.io/docs/minio/linux/administration/object-management.html#object-organization-and-planning and follow those guidelines?

**Reply by opslinux** [reply]
Copy that,Thank you all.

**Reply by opslinux** [reply]
> The same wrong happened to me, any news?

Additional information: file uploads occasionally fail with regularity, minio parameters have been adjusted such as MINIO_STORAGE_WRITE_TIMEOUT=60s MINIO_STORAGE_READ_TIMEOUT=60s MINIO_API_STALE_UPLOADS_EXPIRY=12h MINIO APIREOUESTS MAX_DURATION=60s

**Reply by klauspost** [reply]
If you need us to look into the performance of your setup, reach out on sales@min.io so we can get you set up with a subscription.
